### PR TITLE
[incubator/rundeck] allow configurable nginx conf. Add example

### PR DESCRIPTION
**Pull request by georgekaz**
#### What this PR does / why we need it:
I found it impossible to configure oauth2-proxy in front of rundeck without editing the nginx.conf. I wanted to provide a way to change this via values and make the config templatable. There's an example of how you can configure it (not actually my personal final use case)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: incubator/rundeck/templates/nginx-configmap.yaml [hunk 1]
```diff
@@ -5,4 +5,8 @@ metadata:
 type: Opaque
 data:
   nginx.conf: |-
+{{- if .Values.nginxConfOverride }}
+{{ tpl .Values.nginxConfOverride . | indent 4 }}
+{{- else }}
 {{ .Files.Get "files/nginx/nginx.conf" | indent 4 }}
+{{- end }}
```
Comment (dwardu89) [inline incubator/rundeck/templates/nginx-configmap.yaml]: Can you add a new line please?

**Comment by dwardu89** [comment]
@georgekaz this is great, how about setting this by declaring a config map. Using both approaches this would require us to ensure that we also add a hash to the deployment in order to make the deployment roll a new instance with the new config.

File: incubator/rundeck/templates/nginx-configmap.yaml (hunk 1 above)
Comment (georgekaz) [reply incubator/rundeck/templates/nginx-configmap.yaml]: Sure, that's no problem

**Comment by georgekaz** [comment]
I'm not sure what you mean by "declaring a configmap". This is already a configmap mounted into the deployment. However you're right that it is a problem that the nginx container doesn't reload/restart when that configmap is updated. I'm not sure how to resolve that though.

**Pull request merged**